--- setup.py.old	2007-02-24 16:44:31.000000000 -0800
+++ setup.py	2007-02-24 16:47:43.000000000 -0800
@@ -1,4 +1,6 @@
 # Copyright (c) 2006 Zach Tibbitts ('zachtib') <zach@collegegeek.org>
+# Modified by Peter Gordon ('codergeek42') <peter@thecodergeek.com> to build
+# against a system-installed copy of libtorrent, not the included sources.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -24,20 +26,15 @@
 
 pythonVersion = platform.python_version()[0:3]
 
-removals = ['-g', '-DNDEBUG', '-O2', '-Wstrict-prototypes']
 additions = ['-DNDEBUG', '-O2']
 
 if pythonVersion == '2.5':
 	cv_opt = sysconfig.get_config_vars()["CFLAGS"]
-	for removal in removals:
-		cv_opt = cv_opt.replace(removal, " ")
 	for addition in additions:
 		cv_opt = cv_opt + " " + addition
 	sysconfig.get_config_vars()["CFLAGS"] = ' '.join(cv_opt.split())
 else:
 	cv_opt = sysconfig.get_config_vars()["OPT"]
-	for removal in removals:
-		cv_opt = cv_opt.replace(removal, " ")
 	for addition in additions:
 		cv_opt = cv_opt + " " + addition
 	sysconfig.get_config_vars()["OPT"] = ' '.join(cv_opt.split())
@@ -46,46 +43,11 @@
 
 
 deluge_core = Extension('deluge_core',
-                    include_dirs = ['./libtorrent', './libtorrent/include', './libtorrent/include/libtorrent',
-												'/usr/include/python' + pythonVersion],
-                    libraries = ['boost_filesystem', 'boost_date_time',
-											'boost_program_options', 'boost_regex',
-											'boost_serialization', 'boost_thread', 'z', 'pthread'],
-                    extra_compile_args = ["-Wno-missing-braces"],
-                    sources = ['src/deluge_core.cpp',
-                    					 'libtorrent/src/alert.cpp',
-										 'libtorrent/src/allocate_resources.cpp',
-										 'libtorrent/src/bt_peer_connection.cpp',
-										 'libtorrent/src/entry.cpp',
-										 'libtorrent/src/escape_string.cpp',
-										 'libtorrent/src/file.cpp',
-										 'libtorrent/src/http_tracker_connection.cpp',
-					                	 'libtorrent/src/identify_client.cpp',
-										 'libtorrent/src/ip_filter.cpp',
- 										 'libtorrent/src/peer_connection.cpp',
-						             	 'libtorrent/src/piece_picker.cpp',     
-										 'libtorrent/src/policy.cpp',       
-					                	 'libtorrent/src/session.cpp',   
-					                	 'libtorrent/src/session_impl.cpp',
-                               			 'libtorrent/src/sha1.cpp',
-									     'libtorrent/src/stat.cpp',
-									     'libtorrent/src/storage.cpp',
-								       	 'libtorrent/src/torrent.cpp',
-										 'libtorrent/src/torrent_handle.cpp',
-										 'libtorrent/src/torrent_info.cpp',
-										 'libtorrent/src/tracker_manager.cpp',
-										 'libtorrent/src/udp_tracker_connection.cpp',
-					                	 'libtorrent/src/web_peer_connection.cpp',
-										 'libtorrent/src/kademlia/closest_nodes.cpp',
-										 'libtorrent/src/kademlia/dht_tracker.cpp',
-										 'libtorrent/src/kademlia/find_data.cpp',
-										 'libtorrent/src/kademlia/node.cpp',
-										 'libtorrent/src/kademlia/node_id.cpp',
-										 'libtorrent/src/kademlia/refresh.cpp',
-										 'libtorrent/src/kademlia/routing_table.cpp',
-										 'libtorrent/src/kademlia/rpc_manager.cpp',
-										 'libtorrent/src/kademlia/traversal_algorithm.cpp'])
-
+	include_dirs = [sysconfig.get_python_inc(), '/usr/include', '/usr/include/libtorrent'],
+	libraries = ['boost_filesystem', 'torrent'],
+	extra_compile_args = ["-Wno-missing-braces"],
+	sources = ['src/deluge_core.cpp'])
+	
 data = [('share/deluge/glade',  glob.glob('glade/*.glade')),
         ('share/deluge/pixmaps', glob.glob('pixmaps/*.png')),
         ('share/applications' , ['deluge.desktop']),
@@ -104,6 +66,6 @@
 	packages=['deluge'],
 	package_dir = {'deluge': 'src'},
 	data_files=data,
-    ext_package='deluge',
+	ext_package='deluge',
 	ext_modules=[deluge_core]
 	)
