diff -up ./src/pref.py.old ./src/pref.py
--- ./src/pref.py.old	2008-01-19 20:37:40.000000000 -0800
+++ ./src/pref.py	2008-01-19 20:38:28.000000000 -0800
@@ -130,7 +130,7 @@ if common.windows_check():
     "use_lsd" : False,
     "use_utpex" : True,
     "window_height" : 480,
-    "new_releases" : True,
+    "new_releases" : False,
     "window_maximized" : False,
     "window_pane_position" : -1,
     "window_width" : 640,
diff -up ./src/interface.py.old ./src/interface.py
--- ./src/interface.py.old	2008-01-19 20:38:47.000000000 -0800
+++ ./src/interface.py	2008-01-19 20:40:10.000000000 -0800
@@ -1049,6 +1049,7 @@ window, please enter your password"))
             self.manager.quit()
 
     def new_release_check(self):
+        return
         import socket
         import urllib
         timeout = 5
