Index: tags/deluge-0.5.6.95/src/plugins.py
===================================================================
--- tags/deluge-0.5.6.95/src/plugins.py (revision 2225)
+++ tags/deluge-0.5.6.95/src/plugins.py (revision 2309)
@@ -62,7 +62,11 @@
                     if 'deluge_init' in dir(mod):
                         if modname != "TorrentPieces":
-                            print "Initialising plugin",modname
-                            mod.deluge_init(path)
-                            self.available_plugins[mod.plugin_name] = mod
+                            print "Initialising plugin", modname
+                            try:
+                                mod.deluge_init(path)
+                            except:
+                                print "Cant init plugin", modname
+                            else:
+                                self.available_plugins[mod.plugin_name] = mod
     
     def get_available_plugins(self):
